### PR TITLE
feat: separate setup-token and verify-token into 2 jobs

### PR DESCRIPTION
**Pull request by laurentsimon**
closes https://github.com/slsa-framework/slsa-github-generator/issues/1814

successful run at https://github.com/laurentsimon/slsa-github-generator/actions/runs/4419476426

### CODE REVIEW

File: .github/workflows/scripts/e2e-verify.common.sh [hunk 1]
```diff
@@ -115,7 +115,7 @@ e2e_verify_decoded_token() {
     # GitHub's information.
     _e2e_verify_query "$decoded_token" "$GITHUB_ACTOR_ID" '.github.actor_id'
     _e2e_verify_query "$decoded_token" "$GITHUB_EVENT_NAME" '.github.event_name'
-    _e2e_verify_query "$decoded_token" "$GITHUB_JOB" '.github.job'
+    _e2e_verify_query "$decoded_token" "setup-token" '.github.job'
```
Comment (ianlewis) [inline .github/workflows/scripts/e2e-verify.common.sh:118]: I wonder if we should even record `$GITHUB_JOB` if it's going to be different than the build job. Maybe make an issue for this so we can follow up?

**Review by ianlewis — APPROVED** [review]

File: .github/workflows/scripts/e2e-verify.common.sh (hunk 1 above)
Comment (laurentsimon) [reply .github/workflows/scripts/e2e-verify.common.sh]: I removed the job from the token. Successful run https://github.com/laurentsimon/slsa-github-generator/actions/runs/4429655425

Comment (ianlewis) [reply .github/workflows/scripts/e2e-verify.common.sh]: Looks like the unit tests for `verify-token` need to be updated to remove the job id.

Comment (laurentsimon) [reply .github/workflows/scripts/e2e-verify.common.sh]: They are removed in the latest commit.

Comment (laurentsimon) [reply .github/workflows/scripts/e2e-verify.common.sh]: my bad, let me fix this!

**Pull request merged**